IBX-11485: Update doc for Symfony 7.4 (#3098)
ezsystems/developer-documentation: docs/update_and_migration/from_5.0/update_from_5.0.md

curl -fs https://get.symfony.com/cloud/configurator | bash
```

## v5.0.7

### Update Symfony from 7.3 to 7.4

This version of [[= product_name =]] requires [Symfony 7.4](https://symfony.com/releases/7.4).
Update Symfony constraints in `composer.json` before updating the packages.

1. In `composer.json`, update `extra.symfony.require` to allow installing higher Symfony version:

    ```json
    "extra": {
        "symfony": {
            "require": "7.4.*"
        }
    }
    ```

2. Review your code, configuration, and third-party bundles for Symfony 7.4 compatibility.

3. Update packages by running:

=== "[[= product_name_headless =]]"

    ``` bash
    yarn upgrade @ibexa/frontend-config @ibexa/ts-config
    composer require ibexa/headless:v5.0.7 --with-all-dependencies --no-scripts

>     composer recipes:install ibexa/headless --force -v

Reviewer: due to how flex resolves recipes, and that we forked `JMSTranslationBundle` this command at first installs recipes for our fork - so it tries to add `JMSTranslationBundle` to `bundles.php` which is already there, and then it UNinstall recipes for original repository and that step eventually removes entry from `bundles.php`. There is no easy way around that, that would not affect fresh installs, so our suggestion at this moment is that after recipes:install you should manually restore

`JMS\TranslationBundle\JMSTranslationBundle::class => ['all' => true],` in `bundles.php` in old position

Author: Thanks, added in https://github.com/ibexa/documentation-developer/pull/3098/changes/eb85427802dd2e98db95800be0424c58ca08e283 , could you have a look?

Recipes - love them when they work, hate them when they don't 😓

curl -fs https://get.symfony.com/cloud/configurator | bash
```

## v5.0.7

### Update Symfony from 7.3 to 7.4

This version of [[= product_name =]] requires [Symfony 7.4](https://symfony.com/releases/7.4).
Update Symfony constraints in `composer.json` before updating the packages.

1. In `composer.json`, update `extra.symfony.require` to allow installing higher Symfony version:

    ```json
    "extra": {
        "symfony": {
            "require": "7.4.*"
        }
    }
    ```

> 2. To allow installing Symfony 7.4, update the requirements for `symfony` packages in `composer.json` as in the example below::

Author: ```suggestion
2. To allow installing Symfony 7.4, update the requirements for `symfony` packages in `composer.json` as in the example below:
```

curl -fs https://get.symfony.com/cloud/configurator | bash
```

## v5.0.7

### Update Symfony from 7.3 to 7.4

This version of [[= product_name =]] requires [Symfony 7.4](https://symfony.com/releases/7.4).
Update Symfony constraints in `composer.json` before updating the packages.

1. In `composer.json`, update `extra.symfony.require` to allow installing higher Symfony version:

    ```json
    "extra": {
        "symfony": {
            "require": "7.4.*"
        }
    }
    ```

> 2. To allow installing Symfony 7.4, update the requirements for `symfony` packages in `composer.json` as in the example below:
> 
>     ``` diff
>     -        "symfony/console": "7.3.*",
>     +        "symfony/console": "7.4.*",
>     ```

Reviewer 2: I would try to emphase on that every package must be updated so reader don't do it only for the console as in the example:

```suggestion
2. To allow installing Symfony 7.4, update the requirements for **all** `symfony` packages in `composer.json` as in the example below:

    ``` diff
    -        "symfony/<package>": "7.3.*",
    +        "symfony/<package>": "7.4.*",
    ```
```

> 3. Review your code, configuration, and third-party bundles for Symfony 7.4 compatibility.

Reviewer 2: It seems Rector can't help there, no rule for 7.4 in https://github.com/rectorphp/rector-symfony/tree/main/rules

Maybe we could link to https://github.com/symfony/symfony/blob/7.4/UPGRADE-7.4.md sooner from this section

        yarn upgrade @ibexa/frontend-config @ibexa/ts-config
        composer require ibexa/experience:v5.0.7 --with-all-dependencies --no-scripts
        composer recipes:install ibexa/experience --force -v
        ```
    === "[[= product_name_com =]]"

        ``` bash
        yarn upgrade @ibexa/frontend-config @ibexa/ts-config
        composer require ibexa/commerce:v5.0.7 --with-all-dependencies --no-scripts
        composer recipes:install ibexa/commerce --force -v
        ```

5. Restore manually the entry for `JMSTranslationBundle` in `config/bundles.php` to [its previous position](https://github.com/ibexa/commerce-skeleton/blob/v5.0.6/config/bundles.php#L14):

    ``` hl_lines="2"
        FOS\HttpCacheBundle\FOSHttpCacheBundle::class => ['all' => true],
        JMS\TranslationBundle\JMSTranslationBundle::class => ['all' => true],
        Liip\ImagineBundle\LiipImagineBundle::class => ['all' => true],
    ```

For more details about the new version, see the official Symfony [upgrade instructions](https://github.com/symfony/symfony/blob/7.4/UPGRADE-7.4.md) and [blog posts introducing this release](https://symfony.com/blog/category/living-on-the-edge/8.0-7.4).

> Among the changes are:

Reviewer 2: Merge within section "2. Review your code,…"

curl -fs https://get.symfony.com/cloud/configurator | bash
```

## v5.0.7

### Update Symfony from 7.3 to 7.4

This version of [[= product_name =]] requires [Symfony 7.4](https://symfony.com/releases/7.4).
Update Symfony constraints in `composer.json` before updating the packages.

> 1. In `composer.json`, update `extra.symfony.require` to allow installing higher Symfony version:

Reviewer 3: ```suggestion
1. In `composer.json`, update `extra.symfony.require` to allow installing a higher Symfony version:
```

    ```json
    "extra": {
        "symfony": {
            "require": "7.4.*"
        }
    }
    ```

2. To allow installing Symfony 7.4, update the requirements for `symfony` packages in `composer.json` as in the example below:

    ``` diff
    -        "symfony/console": "7.3.*",
    +        "symfony/console": "7.4.*",
    ```

3. Review your code, configuration, and third-party bundles for Symfony 7.4 compatibility.

    For more details about the new version, see the official Symfony [upgrade instructions](https://github.com/symfony/symfony/blob/7.4/UPGRADE-7.4.md) and [blog posts introducing this release](https://symfony.com/blog/category/living-on-the-edge/8.0-7.4).

>     Among the changes are:

Reviewer 3: ```suggestion
    Key changes include:
```

While correct, the original sounded... stiff.

    - Array-based PHP configuration format

        As part of the [array-based PHP configuration format](https://symfony.com/blog/new-in-symfony-7-4-better-php-configuration), a `config/reference.php` file will be created.

>         You should commit this file to the repository.

Reviewer 3: "should" implies that thay may under certain circumstances not commit the file, like Polish "powinien". "must" would imply no other option.

Author: It's optional, but recommended - it's the wording Symfony uses in https://symfony.com/blog/new-in-symfony-7-4-better-php-configuration#:~:text=PHP%2E-,These,config%2F%22%5D%29%2E

        If you decide to configure it (for example, by setting the `APP_SHARE_DIR` environment variable), review your existing scripts for explicit `var/cache` usage (for example, `rm -rf var/cache`) and decide whether to include `var/share` in the script.

4. Update Ibexa packages by running:

    === "[[= product_name_headless =]]"

        ``` bash
        yarn upgrade @ibexa/frontend-config @ibexa/ts-config
        composer require ibexa/headless:v5.0.7 --with-all-dependencies --no-scripts
        composer recipes:install ibexa/headless --force -v
        ```
    === "[[= product_name_exp =]]"

        ``` bash
        yarn upgrade @ibexa/frontend-config @ibexa/ts-config
        composer require ibexa/experience:v5.0.7 --with-all-dependencies --no-scripts
        composer recipes:install ibexa/experience --force -v
        ```
    === "[[= product_name_com =]]"

        ``` bash
        yarn upgrade @ibexa/frontend-config @ibexa/ts-config
        composer require ibexa/commerce:v5.0.7 --with-all-dependencies --no-scripts
        composer recipes:install ibexa/commerce --force -v
        ```

> 5. Restore manually the entry for `JMSTranslationBundle` in `config/bundles.php` to [its previous position](https://github.com/ibexa/commerce-skeleton/blob/v5.0.6/config/bundles.php#L14):

Reviewer 3: ```suggestion
5. Manually restore the entry for `JMSTranslationBundle` in `config/bundles.php` to [its previous position](https://github.com/ibexa/commerce-skeleton/blob/v5.0.6/config/bundles.php#L14):
```
